### DISCUSSION TITLE
After initiating the logout request, Zitadel did not log out of the login state

### DISCUSSION THREAD

**Discussion by tiantang198635** [zitadel/zitadel#9018] [Q&A]
I am using the Saml protocol. SP initiates a logout request, and Zitadel, as the idp, receives the request and redirects to the SingleLogoutService corresponding to SP. However, why is the idp still logged in?

**Comment by fforootd** [comment]
To make sure I understand this correctly.

The user is still logged in Zitadel after the logout should have happend, correct?

Can you share where you see this and with what version?

**Reply by tiantang198635** [reply]
Is it normal for Zitadel, as an idp, to remain logged in and not logged out after the user initiates logout through SP?
I use the following script for installation："docker compose --file docker-compose-base.yaml --file docker-compose-nginx.yaml up --detach proxy-enabled-tls "
and the version is hrms/dev latest，As shown in the following figure：
![image](https://github.com/user-attachments/assets/edcda996-3acb-408d-8bd0-49acf666a4fe)
and my web url is https://id.appcara.com.hk/

and The following is the logout code for the SP side:
![image](https://github.com/user-attachments/assets/578dd25c-e834-40d6-adce-17fd87e8c8ff)

and The following is the code to receive the idp logout response：
![image](https://github.com/user-attachments/assets/1bca01c8-b981-4969-b204-e25e47f9397a)

the following is the log of logout response:
![image](https://github.com/user-attachments/assets/2e3ae219-6722-42a4-85c3-71bf35fa6284)

May I ask if there are any errors in the code or parameters? thank you！

**Reply by fforootd** [reply]
Of the top of my head I would say we do SLO (Single Logout) with SAML. So it should not be normal.

What version of Zitadel are you using? (Please check the stdout when starting Zitadel and not the docker image tag)

@livio-a @stebenz might know more about this in detail.

**Reply by tiantang198635** [reply]
![image](https://github.com/user-attachments/assets/f16a2225-fbb2-4e6f-9cc6-27443fa968d6)
the version is :v2.64.1

In fact, I used Zitadel's cloud service, but it didn't work. The cloud service address is: https://test-plvkyb.us1.zitadel.cloud/
@livio-a @stebenz @fforootd

**Reply by tiantang198635** [reply]
Excuse me, how should I handle this issue? Do I need to call the Session API? @livio-a @stebenz @fforootd

**Comment by tiantang198635** [comment]
Has anyone encountered this problem before?

**Reply by saugion** [reply]
No solution yet?

**Comment by saugion** [comment]
I think I have the same issue:
* after logging out with Zitadel, I'm redirected to the login page -> OK
* i try to login again, after clicking the idp button i'm redirected to the idp and directly logged in, without inserting email and password -> NO OK

We use the same idp in other applications without zitadel, and when we logged out we are actually logged out from the idp, we saw the SAML logout call as expected.

**Comment by hifabienne** [comment]
This worked as intended, till now.
However, we just released a new Zitadel version where you can configure Federated Logout on your SAML Idp.
https://github.com/zitadel/zitadel/pull/9931

**Reply by saugion** [reply]
Hello @hifabienne, for us the federated logout is still not working. With the last version, the scenario described in my last comment still happens. Even if we manually terminate the session through the API, the slo (saml logout) is not performed.

Version 3.2.0, checkbox Federated Logout Enabled checked

**Reply by hifabienne** [reply]
@livio-a can you have a look at this?

**Reply by saugion** [reply]
Also left a comment here, if it can help https://github.com/zitadel/zitadel/issues/9228#issuecomment-2909189505
